2006-09-17  Robert Millan  <rmh@aybabtu.com>

	* util/i386/pc/biosdisk.c (read_device_map): Don't abort with error if
	realpath fails (e.g. to cope with missing /dev/fd0 when we're aren't
	writing to floppy).

Index: util/i386/pc/biosdisk.c
===================================================================
RCS file: /sources/grub/grub2/util/i386/pc/biosdisk.c,v
retrieving revision 1.12
diff -u -r1.12 biosdisk.c
--- util/i386/pc/biosdisk.c	4 Jun 2006 15:56:55 -0000	1.12
+++ util/i386/pc/biosdisk.c	17 Sep 2006 16:29:39 -0000
@@ -556,7 +556,10 @@
 	 symbolic links.  */
       map[drive] = xmalloc (PATH_MAX);
       if (! realpath (p, map[drive]))
-	grub_util_error ("Cannot get the real path of `%s'", p);
+	{
+	  map[drive] = '\0';
+	  grub_util_info ("Cannot get the real path of `%s'", p);
+	}
 #else
       map[drive] = xstrdup (p);
 #endif
